lib/main_common.pm

```diff
@@ -1545,41 +1546,53 @@ sub load_extra_tests {
     loadtest "console/consoletest_setup";
     loadtest 'console/integration_services' if is_hyperv;
     loadtest "console/hostname";
-    if (any_desktop_is_applicable()) {
+    # Extra tests are too long, split the test into subtest according to the
+    # EXTRATEST variable; to maintain compatibility, run all tests if the
+    # variable is equal 1
+    if (check_var('EXTRATEST', 1) && any_desktop_is_applicable()) {
         load_extra_tests_desktop;
     }
+    elsif (check_var('EXTRATEST', 1)) {
```

Comment:
This code can now run even if `any_desktop_is_applicable()` returns true, it was not supposed to.

Reply:
Well, it would not be called because in line 1552 we checked for `any_desktop_is_applicable()` and here we are in an elsif, right?

Reply:
Sorry, I did not describe it correctly. My concern is about this flow:
```
if (check_var('EXTRATEST', 1) && any_desktop_is_applicable()) ...
elsif (check_var('EXTRATEST', 1)) ...
else ...
```
for ``EXTRATEST==toolkit`` and ``any_desktop_is_applicable() == true`` it will jump to the third branch, which was supposed to be console only (there is ``loadtest "console/consoletest_finish";
``)